### DISCUSSION TITLE
Any plans for .NET 6.0?

### DISCUSSION THREAD

**Discussion by MrMeapz** [jchristn/WatsonWebsocket#76] [Q&A]
I am trying to use this for Unity. So my client is written in .NET Framework and my server is written in .NET 6.0 and running on Debian 11. I tested everything locally on Windows and everything works fine. But once I tried testing if it worked remotely (Windows to Linux) it did not.

![image](https://user-images.githubusercontent.com/32072093/139556258-6bde2be1-572c-4640-971f-44d1849d85b6.png)
According to this issue: https://github.com/dotnet/runtime/issues/34399 : "The implementation of HttpListener on Linux is substantially different than Windows. Windows uses the original .NET Framework (ported to .NET Core) implemented layered on top of http.sys. Linux uses a separate implementation originally from Mono.".

Is it possible that support for .NET 6.0 will ever be a thing? Or perhaps I am doing something wrong?
Windows server is not really an option for me :)

**Comment by jchristn** [comment]
You could clone and modify the .csproj files to include net6.0 and see if that works. I wasn't planning to add it until it was GA.

**Reply by MrMeapz** [reply]
I am testing locally on Windows already, both client and server are in .NET 6.0RC2 and they work fine, as long as both are running locally on Windows. As soon as I move the Server remotely to my Linux server and run it there and try to connect locally with my client to my remote server it no longer works. I guess it is something on the server end that has to do with the operating system it is running on. In short: If I run the server on Windows then I have no errors. If I run this server on Linux I get the error in the screenshot you see in my post above.

**Reply by MrMeapz** [reply]
I currently use Websocketsharp, which does work on Linux. But I am afraid I've hit the maximum capabilities of that library, hence why I wanted to upgrade to Watson Websocket since it is asynchronous.

**Reply by jchristn** [reply]
Cool, let me know if you have any issues with .NET 6.  Like I mentioned I'll add it into the library once .NET 6 is generally available.

**Reply by MrMeapz** [reply]
.NET 6.0 has released today. ;)
3 years of long term support.

https://dotnet.microsoft.com/download/dotnet/6.0

**Reply by jchristn** [reply]
Awesome!  Thanks for letting me know. Just saw the email from Microsoft. Will add it ASAP and report back. Cheers, Joel

**Reply by MrMeapz** [reply]
That's great! I am really looking forward to switching to this awesome library! Keep up the awesome work Joel!

**Reply by jchristn** [reply]
Got you fam :)

NuGet: https://www.nuget.org/packages/WatsonWebsocket/2.3.2.1
Commit: https://github.com/jchristn/WatsonWebsocket/commit/58e18f83a3b355a6a0a9fad124bf1a2af0673c8b

Thanks for your patience, and for the nudge!

**Reply by MrMeapz** [reply]
You're a fast one! Awesome. I'll look at it soon. Thanks a lot man I appreciate it.
